lib/src/sidenav/Sidenav.tsx

```diff
@@ -67,49 +151,148 @@ const SidenavContainer = styled.div<SidenavPropsType>`
+    margin-right: 8px;
+  }
+  box-sizing: border-box;
+  width: 100%;
+
+  font-family: ${(props) => props.theme.groupTitleFontFamily};
+  font-style: ${(props) => props.theme.groupTitleFontStyle};
+  font-weight: ${(props) => props.theme.groupTitleFontWeight};
+  font-size: ${(props) => props.theme.groupTitleFontSize};
+  line-height: 19px;
+
+  display: flex;
+  align-items: center;
+  margin: 0px;
+  padding: 7px 24px;
+`;
+
+const SidenavGroupTitleButton = styled.button`
+  all: unset;
+  cursor: pointer;
+  justify-content: space-between;
+  box-sizing: border-box;
+  width: 100%;
+
+  font-family: ${(props) => props.theme.groupTitleFontFamily};
+  font-style: ${(props) => props.theme.groupTitleFontStyle};
+  font-weight: ${(props) => props.theme.groupTitleFontWeight};
+  font-size: ${(props) => props.theme.groupTitleFontSize};
+  line-height: 19px;
+
+  display: flex;
+  align-items: center;
+  margin: 0px;
+  padding: 7px 24px;
+  &:focus {
+    outline: 2px solid ${(props) => props.theme.linkFocusColor};
+    outline-offset: -2px;
+  }
+  &:hover {
+    background-color: ${(props) => props.theme.groupTitleHoverBackgroundColor};
+  }
+`;
+
+type StyledLinkProps = {
+  selected: boolean;
+};
+const SidenavLink = styled.a<StyledLinkProps>`
   letter-spacing: ${(props) => props.theme.linkFontLetterSpacing};
+
   text-transform: ${(props) => props.theme.linkFontTextTransform};
   text-decoration: ${(props) => props.theme.linkTextDecoration};
-  margin: ${(props) =>
-    `${props.theme.linkMarginTop} ${props.theme.linkMarginRight} ${props.theme.linkMarginBottom} ${props.theme.linkMarginLeft}`};
+
+  font-family: ${(props) => props.theme.linkFontFamily};
+  font-style: ${(props) => props.theme.linkFontStyle};
+  font-weight: ${(props) => props.theme.linkFontWeight};
+  font-size: ${(props) => props.theme.linkFontSize};
+  line-height: 19px;
+
+  ${(props) =>
+    props.selected
+      ? `color: ${props.theme.linkSelectedFontColor}; background: ${props.theme.linkSelectedBackgroundColor};`
+      : `color: ${props.theme.linkFontColor}; background: transparent;`}
+
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+
+  box-shadow: 0 0 0 2px transparent;
+
+  padding: 7px 24px;
+
   cursor: pointer;
+  svg {
+    width: 16px;
+    margin-right: 8px;
+  }
+
+  &:hover {
+    ${(props) =>
+      props.selected
+        ? `color: ${props.theme.linkSelectedHoverFontColor}; background: ${props.theme.linkSelectedHoverBackgroundColor};`
+        : `color: ${props.theme.linkFontColor}; background: ${props.theme.linkHoverBackgroundColor};`}
+  }
 
-  :focus-visible {
+  &:focus {
     outline: 2px solid ${(props) => props.theme.linkFocusColor};
-    outline-offset: 1px;
+    outline-offset: -2px;
+  }
+
+  &:active {
+    color: #ffffff;
+    background: #4d4d4d;
+    outline: 2px solid #0095ff;
+    outline-offset: -2px;
   }
 `;
 
-DxcSidenav.Title = Title;
-DxcSidenav.Subtitle = Subtitle;
+const SidenavSpan = styled.span`
```

Comment:
Try to avoid using the HTML tag in the styled-components naming. Instead, use names that make sense with the context or their placement inside the component. In this case, for example, `SidenavContent` would fit a little bit better.